projects/storefrontlib/cms-components/cart/validation/cart-item-warning/cart-item-validation-warning.component.ts

```diff
@@ -0,0 +1,33 @@
+import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
+import { map } from 'rxjs/operators';
+import { CartValidationStateService } from '../cart-validation-state.service';
+import { ICON_TYPE } from '../../../misc';
+import { CartModification, CartValidationStatusCode } from '@spartacus/core';
+
+@Component({
+  selector: 'cx-cart-item-validation-warning',
+  templateUrl: './cart-item-validation-warning.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class CartItemValidationWarningComponent {
+  @Input()
+  code: string;
+
+  iconTypes = ICON_TYPE;
+  isVisible = true;
+
+  cartModification$ =
+    this.cartValidationStateService.cartValidationResult$.pipe(
+      map((modificationList: CartModification[]) =>
```

Comment:
```suggestion
      map((modificationList) =>
```

Isn't it automatically typed from the service?